spec/view_component/form/select_component_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe ViewComponent::Form::SelectComponent, type: :component do
+  let(:object)  { OpenStruct.new }
+  let(:form)    { form_with(object) }
+  let(:options) { {} }
+  let(:html_options) { {} }
+
+  let(:component) { render_inline(described_class.new(form, object_name, :role, [["Admin", :admin], ["Manager", :manager]], options, html_options)) }
+  let(:component_html_attributes) { component.css("select").first.attributes }
+
+  context "with simple args" do
+    it do
+      expect(component.to_html).to eq(
+        %(<select name="user[role]" id="user_role">) +
+          %(<option value="admin">Admin</option>\n) +
+          %(<option value="manager">Manager</option>) +
+        %(</select>)
+      )
+    end
+  end
+
+  context "with custom html classes" do
+    let(:html_options) { { class: "custom-css-class" } }
+
+    it { expect(component_html_attributes["class"].to_s).to include("custom-css-class") }
+  end
+
+  context "with custom data attributes" do
+    let(:html_options) { { data: { key: "value" } } }
+
+    it { expect(component_html_attributes["data-key"].to_s).to include("value") }
+  end
```

Comment:
Not sure how to reuse shared examples for that... Note that the only difference with the shared examples is that we're using `let(:html_options)` instead of `let(:options)`.

Reply:
Do we have the same case for an other component?

Reply:
Yes: #23, #24

Reply:
Ok Maybe we could add a param to the shared example for that.

https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples

We may use `options_keword_arg_name` or something like that as param, with `:options` as default. What do you think about?

Reply:
Done: b55647d